packages/person/src/person-search/controller.ts

```diff
@@ -0,0 +1,380 @@
+                name: 'No matching person found',
+                isExpired: true,
+              },
+            ];
+          }
+        }
+        // set isSelected on result items
+        this.result = this.setSelected(this.mapPersonSearchResult(result));
+        return this.result;
+      },
+      () => [this.#queryString],
+    );
+  }
+
+  protected updateResolver = (resolver?: PersonResolver): void => {
+    this.resolver = resolver;
+    this.#host.requestUpdate();
+  };
+
+  public hostConnected(): void {
+    requestAnimationFrame(() => {
+      if (this.#host.textInputElement && this.#host.autofocus) {
+        this.#host.textInputElement.focus();
+      }
+    });
+
+    /* Connect provider to controller's updateResolver method */
+    const event = new PersonControllerConnectEvent({
+      detail: {
+        disconnectedCallback: (callback) => {
+          this.disconnectProvider = callback;
+        },
+        updateResolver: this.updateResolver,
+      },
+      bubbles: true,
+      composed: true,
+      cancelable: false,
+    });
+    this.#host.dispatchEvent(event);
+
+    /* add click event to body */
+    document.body.addEventListener('click', this._handleGlobalClick);
+  }
+
+  public hostDisconnected(): void {
+    if (this.disconnectProvider) {
+      this.disconnectProvider();
+    }
+    /* remove global event listeners */
+    document.body.removeEventListener('click', this._handleGlobalClick);
+    document.body.removeEventListener('keyup', this._handleGlobalKeyUp);
+  }
+
+  /**
+   * Close dropdown when click oustside host id
+   */
+  private _handleGlobalClick = (e: MouseEvent): void => {
+    const t = e.target as HTMLElement;
+    if (t && !(this.#host as unknown as HTMLElement).contains(t)) {
+      this.isOpen = false;
+    }
+  };
+
+  /**
+   * Close dropdown on escape key.
+   * only called when isOpen === true
+   */
+  private _handleGlobalKeyUp = (e: KeyboardEvent): void => {
+    /* Close on Escape */
+    if (e.key === 'Escape') {
+      if (this.#host.textInputElement) {
+        /* Close element */
+        this.isOpen = false;
+      }
+    } else if (e.key === 'Tab') {
+      if (document.activeElement?.id !== this.#host.id) {
+        this.isOpen = false;
+      }
+    }
+  };
+
+  /**
+   * Mutates result to set parameters like isSelected.
+   * @param result SearchableDropdownResult
+   * @returns result
+   */
+  private mapPersonSearchResult(result: PersonQueryDetails): SearchableDropdownResult {
+    return result.map((item) => {
+      const person = {
+        id: item.azureUniqueId,
+        title: item.name,
+        subTitle: item.mail,
+        // graphic: item.azureUniqueId.match(/[a-z0-9-]/) ? item.azureUniqueId : undefined, // to only load avatar on real results
+        graphic: item.mail ?? undefined, // to only load avatar on real results
+        isDisabled: item.isExpired,
+        isError: item.azureUniqueId === 'error',
+      };
+      return person;
+    });
+  }
```

Comment:
extract function to a `utils.ts` and export this function
this function only maps, is not dependent on the actual class.
by extracting the function, the node runnner can L1 cache this function instead of having one for each instance